arch/xtensa/src/esp32s3/esp32s3_spi.h

```diff
@@ -0,0 +1,170 @@
+
+#ifndef __ARCH_XTENSA_SRC_ESP32S3_ESP32S3_SPI_H
+#define __ARCH_XTENSA_SRC_ESP32S3_ESP32S3_SPI_H
+
+/****************************************************************************
+ * Included Files
+ ****************************************************************************/
+
+#include <nuttx/config.h>
+
+#ifndef __ASSEMBLY__
+
+#undef EXTERN
+#if defined(__cplusplus)
+#define EXTERN extern "C"
+extern "C"
+{
+#else
+#define EXTERN extern
+#endif
+
+#ifdef CONFIG_ESP32S3_SPI
+
+#include <nuttx/spi/spi.h>
+
+#ifdef CONFIG_ESP32S3_SPI2
+#  define ESP32S3_SPI2 2
+#endif
+
+/****************************************************************************
+ * Public Function Prototypes
+ ****************************************************************************/
+
+/****************************************************************************
+ * Name: esp32s3_spibus_initialize
+ *
+ * Description:
+ *   Initialize the selected SPI bus.
+ *
+ * Input Parameters:
+ *   port     - Port number (for hardware that has multiple SPI interfaces)
+ *
+ * Returned Value:
+ *   Valid SPI device structure reference on success; NULL on failure
+ *
+ ****************************************************************************/
+
+struct spi_dev_s *esp32s3_spibus_initialize(int port);
+
+/****************************************************************************
+ * Name:  esp32s3_spi[0|1]_select and esp32s3_spi[0|1]_status
+ *
+ * Description:
+ *   The external functions, esp32s3_spi[0|1]_select,
+ *   esp32s3_spi[0|1]_status, and esp32s3_spi[0|1]_cmddata must be provided
+ *   by board-specific logic.
+ *   These are implementations of the select, status, and cmddata methods of
+ *   the SPI interface defined by struct spi_ops_s (include/nuttx/spi/spi.h).
+ *   All other methods (including esp32s3_spibus_initialize()) are provided
+ *   by common ESP32S3 logic. To use this common SPI logic on your board:
+ *
+ *   1. Provide logic in esp32s3_board_initialize() to configure SPI chip
+ *      select pins.
+ *   2. Provide esp32s3_spi[0|1]_select() and esp32s3_spi[0|1]_status()
+ *      functions in your board-specific logic. These functions will perform
+ *      chip selection and status operations using GPIOs in the way your
+ *      board is configured.
+ *   3. If CONFIG_SPI_CMDDATA is defined in your NuttX configuration file,
+ *      then provide esp32s3_spi[0|1]_cmddata() functions in your
+ *      board-specific logic. These functions will perform cmd/data selection
+ *      operations using GPIOs in the way your board is configured.
+ *   4. Add a call to esp32s3_spibus_initialize() in your low level
+ *      application initialization logic.
+ *   5. The handle returned by esp32s3_spibus_initialize() may then be used
+ *      to bind the SPI driver to higher level logic (e.g., calling
+ *      mmcsd_spislotinitialize(), for example, will bind the SPI driver to
+ *      the SPI MMC/SD driver).
```

Comment:
```suggestion
 * Name:  esp32s3_spi[2|3]_select and esp32s3_spi[2|3]_status
 *
 * Description:
 *   The external functions, esp32s3_spi[2|3]_select,
 *   esp32s3_spi[2|3]_status, and esp32s3_spi[2|3]_cmddata must be provided
 *   by board-specific logic.
 *   These are implementations of the select, status, and cmddata methods of
 *   the SPI interface defined by struct spi_ops_s (include/nuttx/spi/spi.h).
 *   All other methods (including esp32s3_spibus_initialize()) are provided
 *   by common ESP32-S3 logic. To use this common SPI logic on your board:
 *
 *   1. Provide logic in esp32s3_board_initialize() to configure SPI chip
 *      select pins.
 *   2. Provide esp32s3_spi[2|3]_select() and esp32s3_spi[2|3]_status()
 *      functions in your board-specific logic. These functions will perform
 *      chip selection and status operations using GPIOs in the way your
 *      board is configured.
 *   3. If CONFIG_SPI_CMDDATA is defined in your NuttX configuration file,
 *      then provide esp32s3_spi[2|3]_cmddata() functions in your
 *      board-specific logic. These functions will perform cmd/data selection
 *      operations using GPIOs in the way your board is configured.
 *   4. Add a call to esp32s3_spibus_initialize() in your low level
 *      application initialization logic.
 *   5. The handle returned by esp32s3_spibus_initialize() may then be used
 *      to bind the SPI driver to higher level logic (e.g., calling
 *      mmcsd_spislotinitialize(), for example, will bind the SPI driver to
 *      the SPI MMC/SD driver).
```
Two issues here:
1) Change [0|1] to [2|3] to make documentation consistent.
2) Missing hyphen in **ESP32-S3**.